tf5muxserver/mux_server.go

```diff
@@ -35,6 +35,9 @@ type muxServer struct {
 	// Resource capabilities are cached during GetMetadata/GetProviderSchema
 	resourceCapabilities map[string]*tfprotov5.ServerCapabilities
 
+	// Routing for resource identity
+	resourceIdentity map[string]tfprotov5.ProviderServer
```

Comment:
Interesting 🤔 , I'm wondering if we should just use the `muxServer.resources` for routing resource identity RPC calls (currently just UpgradeResourceIdentity and then let the downstream server return an error if there is no resource identity (since it's an optional implementation).

If we do that, we can remove this caching here and the related discovery code for "resource identity servers", since (as noted somewhere in a comment in this PR), the mux server that implements the resource, will also implement the identity.

Reply:
(it's not guaranteed that `GetProviderSchema` will be called by external programs before `GetResourceIdentitySchemas`, but the `s.getResourceServer` function I believe will call downstream `GetMetadata` RPCs for server discovery if that hasn't been populated yet, so we should be covered there)

Reply:
sounds good! I'll add a commit to simplify this 👍

Reply:
done!